### DISCUSSION TITLE
Restore NotificationHandle

### DISCUSSION THREAD

**Discussion by schrieveslaach** [hoodie/notify-rust#196] [General]
Sometimes a notification is created by a process that after creating the notification terminates. For example, starship.rs can create desktop notifications if the command execution took longer (see https://starship.rs/config/#time).

Unfortunately, it is impossible to update the notification if the process terminated. I think it is due to lost dbus connection or handle. But this may lead to many desktop notifications.

Is there a way to restore the `NotificationHandle` (e.g. via `Notification::id`) to update already existing desktop notifications? In the case of starship it may be restored on the basis of [`STARSHIP_SESSION_KEY`](https://starship.rs/config/#logging).

**Comment by schrieveslaach** [comment]
Here is a screenshot of how it looks like:

![grafik](https://github.com/hoodie/notify-rust/assets/16558417/aba3d09b-99b6-4eb7-9ed8-d54e39946d59)

**Comment by hoodie** [comment]
That is an excellent question. If I remember correctly it used to be that e.g. KDE and gnome handled updating notifications slightly differently. Where KDE did allow you to update the notification from another process Gnome required that you hold the same exact connection open. This maybe a bit underspecified. The question whether or not a connection object can be serialized or passed via some IPC principle is something I'd like to ask @zeenix .

Besides that, something that I've seen is that on macOS sending notifications is sometimes done by a second application, that could then run like a service and handle the connections with a longer lifetime. Maybe that is something we could add to toastify.
